protocol.html

```diff
@@ -559,6 +572,40 @@ <h3 property="schema:name">Resource Containment</h3>
 
                   <p><span about="" id="server-basic-container" rel="spec:requirement" resource="#server-basic-container"><span property="spec:statement">The representation and behaviour of containers in Solid corresponds to LDP Basic Container and <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> be supported by <span rel="spec:requirementSubject" resource="spec:Server">server</span>.</span></span> [<a href="https://github.com/solid/specification/issues/47#issuecomment-561675764" rel="cito:citesAsSourceDocument">Source</a>]</p>
+
+                  <section id="authoritative-resource-data" inlist="" rel="schema:hasPart" resource="#authoritative-resource-data">
+                    <h4 property="schema:name">Authoritative Resource Data</h4>
+                    <div datatype="rdf:HTML" property="schema:description">
+                      <p>Container descriptions are not limited to containment triples. To further support client navigation and application interaction, servers can include <a href="#authoritative-information">authoritative information</a> about contained resources as part of the container description.</p>
+
+                      <p><span about="" id="authoritative-contained-resource-data" rel="spec:requirement" resource="authoritative-contained-resource-data"><span property="spec:statement"><span rel="spec:requirementSubject" resource="spec:Server">Servers</span> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> include the following information about contained resources as part of the container description, unless that information does not exist or is expensive to determine:</span></span></p>
+
+                      <dl>
+                        <dt><code>rdf:type</code></dt>
+                        <dd>A class whose URI formed by concatenating <code>http://www.w3.org/ns/iana/media-types/</code>, the IANA Media Type [<cite><a class="bibref" href="#bib-iana-media-types">IANA-MEDIA-TYPES</a></cite>] of the resource, and <code>#Resource</code>.</dd>
+                        <dt><code>stat:size</code></dt>
+                        <dd>A non-negative integer giving the size of the resource in bytes.</dd>
+                        <dt><code>dcterms:modified</code></dt>
+                        <dd>The time when the resource was last modified.</dd>
+                        <dt><code>stat:mtime</code></dt>
+                        <dd>The Unix time when the resource was last modified.</dd>
```

Comment:
This is incorrect. 

NSS does not say when the resource was last modified, as can be seen for example at https://kjetilktest.solidcommunity.net/public/ , prefixes omitted for brevity:

```turtle
@prefix : <#>.
@prefix dct: <http://purl.org/dc/terms/>.
@prefix ldp: <http://www.w3.org/ns/ldp#>.
@prefix stat: <http://www.w3.org/ns/posix/stat#>.
@prefix xsd: <http://www.w3.org/2001/XMLSchema#>.
@prefix pub: <>.
@prefix test: <test/>.

pub:
    a ldp:BasicContainer, ldp:Container;
    dct:modified "2021-10-26T09:56:26Z"^^xsd:dateTime;
    ldp:contains test:;
    stat:mtime 1635242186.156;
    stat:size 4096.
test:
    a ldp:BasicContainer, ldp:Container, ldp:Resource;
    dct:modified "2021-10-26T10:04:40Z"^^xsd:dateTime;
    stat:mtime 1635242680.624;
    stat:size 4096.
```

As can be seen from the most recent timestamp, this resource was updated at the latest at 2021-10-26T10:04:40Z. This resource is given by the `@prefix pub: <>.`, which has an older timestamp.

In reality, NSS sets the last modification to the time determined by inspecting the inode of the underlying filesystem as defined by POSIX. This is *not* the last modification time of the resource, as evident from the above. This can also be verified from the NSS source code: https://github.com/solid/node-solid-server/blob/693e7a15d5cae3d4b4f328ef1810ee6c4ecb7efd/lib/ldp-container.js#L117-L126

The same thing can be seen by various other methods, but this is the simplest illustration I know of. 

So, `stat:mtime` and `dct:modified` needs to be defined in different terms. I guess we could bring in POSIX as a normative reference, but it is not an open standard. I'm not sure how we'd resolve this, actually, that's why I have tried to point out this problem numerous times...

Reply:
Does this mean that an agent cannot otherwise use these predicate in the RDF description of a container resource?

Reply:
To try to be a little more constructive here, there is a possibility to say something like 

"The time where the non-authoritative parts of the representation of the resource was modified."

That covers a bit more than it needs to, but there should be no harm to that.

Reply:
@kjetilk (partial response):
In your example with the NSS behaviour, the description of public/ including authoritative data about itself (the container) is not covered by the requirement in this PR.

What's requested is a requirement on authoritative data about contained resources in the description of a container (this PR). If I overlooked the call to have authoritative data about the container as well in the container description, can someone point me to it?

Reply:
(partial response continued..)

Although this is not explicit (but we can make it so..), recipient observing `dcterms:modified` may interpret/expect that the value would correspond with the `Last-Modified` header value in the response of the contained resource, if the recipient were to request it at that time. Perhaps of minor importance but the `Last-Modified` header remains as a SHOULD (as is from RFC 7232): https://github.com/solid/specification/issues/154 .

Reply:
NSS behaviour seems to match POSIX. `mkdir foo; mkdir foo/bar; mkdir foo/bar/baz; touch foo/bar/baz/qux;` (with a second in between each command just to spot time diffs with `stat FILE`). NSS is just using `stat.mtime` after all.

Reply:
So you are saying that what NSS and CSS presumably do  works because it gives a sense of when the directory was changed in terms o the set of things contained, which does not propagate, which is important.  But it has  formally the problem that if we look at when the content you get from a GET has changed, then for example a client would assume that it does not have to refresh its cache of the container resource because it has a copy of that date.    When in fact the GET contents would have changed because the size or date of a contained resource have changed.  Yes, Kjetil, we must explain that it is the last modified dat of the structure which does not change with the metadata about the contents.  To avoid change propagation all the way to the top with every patch anywhere in the tree.  It is the Structure Last Modified time.

Reply:
First, @csarven , it is not just the self-description that is the issue here, that `dct:modified` does not reflect the resource modification time and thus is different from the value of `Last-Modified` can also be verified with for example
```sh
curl -D - -H "Accept: text/turtle" https://kjetilktest.solidcommunity.net/
```

As I believe caching is extremely important to the success of Solid, anything that can get in conflict with that is a red flag for me. I would much prefer that whenever we define what the value of `dct:modified` should be that we reference the definition of `Last-Modified` in [Section 2.2 of RFC7232](https://datatracker.ietf.org/doc/html/rfc7232#section-2.2) (but allow for greater precision).

I have a proposal that I'd like to implement that would show how this can be done, that would enable last-modified tracking for all resources, and wouldn't have the problem of cascading. I don't think I should do that right now, because I believe we urgently need to publish 0.9, but I would be sleeping better at night (literally!) if we could leave the option open to do that later. 

For 0.9, I believe we can get where we need to be if we just make it clear that for now, `dct:modified` and `stat:mtime` does not reflect the last modification for the resource, just some of the data. Calling that "Structure Last Modified time" works for me.

Reply:
Good that we share the same interests/care on caching (e.g., "Caching, caching, caching" in the first comment in issue 227 on 2021-02.)

And tracking down TAG findings to communicate/apply ideas:

>For example, Last-Modified is used for representation "metadata". dcterms:modified is for RDF "data".

or what I wrote just two comments ago above about the relationship between the two.

>it is not just the self-description that is the issue here

Well aware. The examples in 227 with the headers served just fine. The self-description example here wasn't necessary or at least seemed confusing. I had to rule it out.

---

I wanted to get to the relevant 7232 bit:

> How that value is
   determined for any given resource is an implementation detail beyond
   the scope of this specification.

We have more freedom in explaining how the Last-Modified value is determined for the container than redefining dcterms:modified (or am I misunderstanding you? or is this effectively the same in the end?)

---

I think we should drop stat:mtime for 1.0.

(I don't see an issue with a server basing Last-Modified off mtime. That's fine by 7232.)

Reply:
Clarified container's Last-Modified and how dcterms:modified corresponds with the Last-Modified of a contained resource: https://github.com/solid/specification/pull/352/commits/4eaf3456c4a8d90a2fb9fd931a8135733914da91 , https://github.com/solid/specification/pull/352/commits/30eee02aafb2077ee48c7ddbe1c596554184f3b8

Deferring to https://github.com/solid/specification/issues/343 for further changes to these requirements.